restaurant/views.py

```diff
@@ -0,0 +1,168 @@
+from rest_framework.decorators import api_view, permission_classes
+from .models import Restaurant, Ticket, Purchase
+from .serializers import RestaurantSerializer, TicketSerializer, TicketUpdateSerializer, PurchaseSerializer
+from user_profile.models import Profile
+from rest_framework.permissions import IsAuthenticated, AllowAny
+from rest_framework.response import Response
+from rest_framework import status
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from user_profile.utils.profile_helpers import get_profile_from_api_request
+from django.db.models import F
+
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def create_restaurant(request):
+  data = request.data
+  user = get_profile_from_api_request(request=request)
+  data.update({'owner': user.id})
+  serializer = RestaurantSerializer(data=data)
+  serializer.is_valid(raise_exception=True)
+  serializer.save()
+
+  return Response(data=serializer.data, status=status.HTTP_201_CREATED)
+
+
+@api_view(['GET'])
+@permission_classes([IsAuthenticated])
+def get_restaurants(request):
```

Comment:
Could you please replace this with ModelViewSet?
Please try to do this as much as you can.,

Reply:
Thanks, let me do that

Reply:
> Could you please replace this with ModelViewSet?
> Please try to do this as much as you can.,

I have implemented on this view